vividus-plugin-web-app-to-rest-api/src/test/java/org/vividus/ssllabs/SslLabsClientTests.java

```diff
@@ -0,0 +1,195 @@
+    {
+        testPerformSslScanWithUnexpectedStatus(HttpStatus.SC_SERVICE_UNAVAILABLE);
+    }
+
+    @Test
+    void shouldPerformSslScanWithUnsupportedStatusCode() throws IOException
+    {
+        testPerformSslScanWithUnexpectedStatus(HttpStatus.SC_BAD_REQUEST);
+    }
+
+    @Test
+    void shouldPerformSslScanWithErrorStatus() throws IOException
+    {
+        try (var grade = mockStatic(Grade.class))
+        {
+            String response =
+                    "{\"host\":\"https://example.com\",\"status\":\"ERROR\",\"statusMessage\":\"Unexpected error\"}";
+            stubResponse(response);
+            assertEquals(Optional.empty(), client.performSslScan(ANALYZED_HOST));
+            grade.verify(() -> Grade.fromString(anyString()), never());
+            assertThat(logger.getLoggingEvents(), is(List.of(
+                    error("Status message '{}' received for host {}", "Unexpected error", "https://example.com"),
+                    errorLogEntry)));
+        }
+    }
+
+    @Test
+    void shouldPerformSslScanWhenReadyStatusDidNotReceived() throws IOException
+    {
+        try (var grade = mockStatic(Grade.class))
+        {
+            String response = "{\"status\":\"\"}";
+            stubResponse(response);
+            assertEquals(Optional.empty(), client.performSslScan(ANALYZED_HOST));
+            grade.verify(() -> Grade.fromString(anyString()), never());
+            assertThat(logger.getLoggingEvents(), hasItems(errorLogEntry));
+        }
+    }
+
+    @Test
+    void shouldPerformSslScanEmptyEndpoints() throws IOException
+    {
+        try (var grade = mockStatic(Grade.class))
+        {
+            String response = "{\"status\":\"READY\",\"endpoints\":[]}";
+            stubResponse(response);
+            assertEquals(Optional.empty(), client.performSslScan(ANALYZED_HOST));
+            grade.verify(() -> Grade.fromString(anyString()), never());
+            assertThat(logger.getLoggingEvents(), is(List.of()));
+        }
+    }
+
+    private void testPerformSslScanWithUnexpectedStatus(int statusCode) throws IOException
+    {
+        try (var grade = mockStatic(Grade.class))
+        {
+            String response = "{\"status\":\"ERROR\"}";
+            HttpResponse httpResponse = new HttpResponse();
+            httpResponse.setStatusCode(statusCode);
+            httpResponse.setResponseBody(response.getBytes(StandardCharsets.UTF_8));
+            when(httpClient.doHttpGet(any(URI.class))).thenReturn(httpResponse);
+            assertEquals(Optional.empty(), client.performSslScan(ANALYZED_HOST));
+            grade.verify(() -> Grade.fromString(anyString()), never());
+            assertThat(logger.getLoggingEvents(), hasItems(
+                    warn("Unexpected status code received: {}\nResponse body: {}", statusCode, response),
+                    errorLogEntry));
+        }
+    }
+
+    private void stubResponse(String response) throws IOException
```

Comment:
```suggestion
    private void mockResponse(String response) throws IOException
```
https://martinfowler.com/articles/mocksArentStubs.html